### PR TITLE
[api-extractor] Fix api-extractor generator to only output lower cases of directives

### PR DESCRIPTION
**Pull request by Claudiazhaoya**
## Summary

This PR aims to fix the output of report by api-extractors to only have lower cased directives.

### CODE REVIEW

File: apps/api-extractor/src/generators/ApiReportGenerator.ts [hunk 1]
```diff
@@ -60,10 +60,10 @@ export class ApiReportGenerator {
     // Emit the triple slash directives
     for (const typeDirectiveReference of Array.from(collector.dtsTypeReferenceDirectives).sort()) {
       // https://github.com/microsoft/TypeScript/blob/611ebc7aadd7a44a4c0447698bfda9222a78cb66/src/compiler/declarationEmitter.ts#L162
-      writer.writeLine(`/// <reference types="${typeDirectiveReference}" />`);
+      writer.writeLine(`/// <reference types="${typeDirectiveReference.toLowerCase()}" />`);
```
Comment (D4N14L) [inline apps/api-extractor/src/generators/ApiReportGenerator.ts:63]: Add comments on both as to why the lower casing is needed

Comment (D4N14L) [reply apps/api-extractor/src/generators/ApiReportGenerator.ts:63]: Are we sure this is the correct fix? Looking online (ex. https://stackoverflow.com/questions/37617325/typing-references-in-typescript-are-case-sensitive) it seems that this field is case-insensitive on Windows, but case-sensitive at runtime on Linux, so ideally we would want to include the exact correct casing here. We could end up actually breaking more people than we fix if this is in-fact the case.

Comment (iclanton) [reply apps/api-extractor/src/generators/ApiReportGenerator.ts:63]: Here's some code in Heft that corrects the casing of a folder path: https://github.com/microsoft/rushstack/blob/a13865bef9a20dab28c044be3504c7326bfe94b1/apps/heft/src/configuration/HeftConfiguration.ts#L170-L175

We could do something similar here.

**Comment by Claudiazhaoya** [comment]
The issue has been solved with some other circumvent solution

**Pull request closed without merging**